src/gluonts/model/deepvar_hierarchical/_estimator.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import List, Optional
+
+import numpy as np
+from mxnet.gluon import HybridBlock
+import mxnet as mx
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.model.deepvar import DeepVAREstimator
+from gluonts.mx.model.predictor import Predictor
+from gluonts.mx.model.predictor import RepresentableBlockPredictor
+from gluonts.mx.distribution import DistributionOutput
+from gluonts.mx.trainer import Trainer
+from gluonts.mx.util import copy_parameters
+from gluonts.time_feature import TimeFeature
+from gluonts.transform import Transformation
+
+
+# Relative imports
+from ._network import (
+    DeepVARHierarchicalPredictionNetwork,
+    DeepVARHierarchicalTrainingNetwork,
+)
+
+
+def projection_mat(S, return_constraint_mat: bool = False):
```

Comment:
Type of S? Could you also add return types to the function declaration rather than in the docstring? I think this is overall the more consistent approach and also helps because it flares up in an IDE if violated.

Reply:
Yes, forgot to add..